### PR TITLE
Major Refactor

### PR DESCRIPTION
**Pull request by leoric-crown**
# Major Refactor
### In this PR I introduced the following changes:
* Added environment variables to `nodemon.json`
* Completely using `async / await` instead of `then()`
* Added Controllers for Collections
* Added User Signup and Login
## TODO: 
- [x] User Authentication
- [ ] CC
- [ ] QA
- [ ] Ready For Deployement
## Further Considerations
Consider which routes need to be protected and how:
/encounters/
DM visible and accessible only except for {status: active} encounters, which players should be able to join or subscribe to with one of their characters.

/characters/
Players should only have access to their own Characters.

/initiatives/
Players should only have access to their own Initiatives.

/users/
Only DM should be able to see a list of Users and their characters.

Open to thoughts and suggestions.

### CODE REVIEW

File: dnd/api/models/user.model.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const mongoose = require('mongoose')
+
+const userSchema = mongoose.Schema({
+    _id: mongoose.Schema.Types.ObjectId,
+    email: {
+        type: String,
+        required: true,
+        unique: true, 
+        match: /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/
```
[the remark this answers is not in the document]
Comment (mquijanogv) [reply dnd/api/models/user.model.js]: actually nvm, not sure if  it would work in a mongoose schema

File: dnd/config/main.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
```
Comment (mquijanogv) [inline dnd/config/main.js]: In config file is where we make use on `ENV` variables. Something like this:
```
 host: process.env.HOST ,
  port: process.env.PORT ,
  dbpath: process.env.DB_PATH ,
  jwt_key: process.env.JWT 
```

File: dnd/api/models/user.model.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (leoric-crown) [reply dnd/api/models/user.model.js]: Indeed I've been looking into it and match: only takes regex

File: dnd/config/main.js (hunk 2 above)
Comment (leoric-crown) [reply dnd/config/main.js]: Care to elaborate? In the end wouldn't be inputting the strings anyway?

File: dnd/api/controllers/initiatives.controller.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
+const config = require('../../config/main')
+const mongoose = require('mongoose')
+const Initiative = require('../models/initiative.model')
+const endpoint = `http://${config.host}:${config.port}/initiatives/`
+const encounterEndpoint = `http://${config.host}:${config.port}/encounters/`
+const characterEndpoint = `http://${config.host}:${config.port}/characters/`
+
```
[the remark this answers is not in the document]
Comment (mquijanogv) [reply dnd/api/controllers/initiatives.controller.js]: It's just to keep consistency across files

**Pull request merged**